### PR TITLE
Add sync Create() method to PublisherClient and SubscriberClient

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

**Review by amanda-tarafa — APPROVED** [review]
LGTM. (Maybe add a couple of tests?)
And also best if @jskeet takes a look as well :) since I'm not that familiar with this library.

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.cs [hunk 1]
```diff
@@ -209,13 +209,35 @@ internal void Validate()
...
+        /// </summary>
+        /// <param name="topicName">The <see cref="TopicName"/> to publish messages to.</param>
+        /// <param name="clientCreationSettings">Optional. <see cref="ClientCreationSettings"/> specifying how to create
+        /// <see cref="PublisherServiceApiClient"/>s.</param>
+        /// <param name="settings">Optional. <see cref="Settings"/> for creating a <see cref="PublisherClient"/>.</param>
+        /// <returns>A <see cref="PublisherClient"/> instance associated with the specified <see cref="TopicName"/>.</returns>
+        public static Task<PublisherClient> CreateAsync(TopicName topicName, ClientCreationSettings clientCreationSettings = null, Settings settings = null) =>
+            CreateMaybeAsync(topicName, clientCreationSettings, settings, isAsync: true);
+
+        private static async Task<PublisherClient> CreateMaybeAsync(TopicName topicName, ClientCreationSettings clientCreationSettings, Settings settings, bool isAsync)
```
Comment (jskeet) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.cs:240]: Definitely worth a comment - presumably the intention is that if "isAsync" is false, this completes synchronously and returns an already-completed (or faulted) task?

**Review by jskeet — APPROVED** [review]
I like the general approach - would like to see a comment on each of the methods though.

Not sure how to test this in a useful way, given that whether or not it completes synchronously will depend on the default credentials.

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.cs (hunk 1 above)
Comment (chrisdunelm) [reply apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.cs:248]: Comments added, and yes that's the intention of using "isAsync"

**Review by jskeet — APPROVED** [review]
Looks good to me. @amanda-tarafa was there anything specific that you think it's worth testing?

**Comment by chrisdunelm** [comment]
OK, I'll merge this now, can add anything extra in a further PR.

**Comment by amanda-tarafa** [comment]
> Looks good to me. @amanda-tarafa was there anything specific that you think it's worth testing?

No, that's fine.

**Pull request merged**